landlab/components/concentration_tracker/concentration_tracker_for_diffusion.py

```diff
@@ -0,0 +1,390 @@
+
+        # use setter for C_w defined below
+        self.C_w = concentration_from_weathering
+
+        # Sediment property concentration field (at links, to calculate dQCdx)
+        self._C_links = np.zeros(self._grid.number_of_links)
+
+        # Sediment property mass field (at links, to calculate dQCdx)
+        self._QC_links = np.zeros(self._grid.number_of_links)
+
+    @property
+    def C_init(self):
+        """Initial concentration in soil/sediment (kg/m^3)."""
+        return self._C_init
+
+    @property
+    def C_br(self):
+        """Concentration in bedrock (kg/m^3)."""
+        return self._C_br
+
+    @property
+    def C_w(self):
+        """Concentration from the weathering process (kg/m^3)."""
+        return self._C_w
+
+    @C_init.setter
+    def C_init(self, new_val):
+        if np.any(new_val < 0.0):
+            raise ValueError("Concentration cannot be negative")
+        self._C_init = return_array_at_node(self._grid, new_val)
+
+    @C_br.setter
+    def C_br(self, new_val):
+        if np.any(new_val < 0.0):
+            raise ValueError("Concentration in bedrock cannot be negative")
+        self._C_br = return_array_at_node(self._grid, new_val)
+
+    @C_w.setter
+    def C_w(self, new_val):
+        if new_val is None:
+            new_val = self._C_br
+        if np.any(new_val < 0.0):
+            raise ValueError("Concentration cannot be negative")
+        self._C_w = new_val
+
+    def concentration(self, dt):
+        """Calculate change in concentration for a time period 'dt'.
+
+        Parameters
+        ----------
+        dt: float (time)
+            The imposed timestep.
+        """
+
+        # Define concentration at previous timestep
+        C_old = self._concentration.copy()
+
+        # Map concentration from nodes to links (following soil flux direction)
+        # Does this overwrite fixed-value/gradient links?
+        self._C_links = map_value_at_max_node_to_link(
+            self._grid, "topographic__elevation", "sediment_property__concentration"
+        )
+        # Replace values with zero for all INACTIVE links
+        self._C_links[self._grid.status_at_link == LinkStatus.INACTIVE] = 0.0
+
+        # Calculate QC at links (sediment flux times concentration)
+        self._qc_at_link = self._grid.at_link["soil__flux"] * self._C_links
+
+        # Calculate flux concentration divergence
+        dQCdx = self._grid.calc_flux_div_at_node(self._qc_at_link)
+
+        # Calculate other components of mass balance equation
+        is_soil = self._soil__depth > 0.0
+
+        old_depth_over_new = np.divide(
+            self._soil__depth_old, self._soil__depth, where=is_soil
+        )
+        old_depth_over_new[~is_soil] = 0.0
+
+        dt_over_depth = np.divide(dt, self._soil__depth, where=is_soil)
+        dt_over_depth[~is_soil] = 0.0
+
+        C_local = C_old * old_depth_over_new
+        C_from_weathering = np.divide(
```

Comment:
Are you ok with changing these to `c_local` and `c_old`? Variable names should be *snake_case*. Notice that GitHub highlights them as though they are classes because of the capital *C*.

Reply:
I have changed all "C" to "conc" across the board for consistency.

Reply:
Perfect. Thank you.